sql/updates/latest-dev.sql

```diff
@@ -63,3 +63,21 @@ CREATE INDEX continuous_aggs_materialization_invalidation_log_idx ON
     _timescaledb_catalog.continuous_aggs_materialization_invalidation_log (
         materialization_id, lowest_modified_value ASC);
 GRANT SELECT ON  _timescaledb_catalog.continuous_aggs_materialization_invalidation_log TO PUBLIC;
+
+-- Suspend any running retention policies that conflict with continuous aggs
+DO $$
+DECLARE
+  jobid INTEGER;
+BEGIN
+    FOR jobid IN
+        SELECT c.id
+        FROM _timescaledb_config.bgw_job a
+        LEFT JOIN _timescaledb_catalog.continuous_agg b ON a.hypertable_id = b.mat_hypertable_id
+        INNER JOIN _timescaledb_config.bgw_job c ON c.hypertable_id = b.raw_hypertable_id
+        WHERE a.proc_name = 'policy_refresh_continuous_aggregate' AND c.proc_name = 'policy_retention' AND c.scheduled
+            AND (a.config->'start_offset')::text::interval > (c.config->'drop_after')::text::interval
```

Comment:
we could have integer offsets as well for cont. aggregates that use integer time buckets.

Reply:
Actually it does work here (I tested this case).  Casting the int to text and then to interval converts the int to a seconds interval, which will do the right thing for determining if the cagg and retention policy overlap.

Reply:
If this works as intended, I suggest a comment to make it clear that the behavior is intentional.

Reply:
Do we need to check the `cascade_to_materializations` settings of the policy prior to being converted to the new format? I guess if that parameter was set to true, the policy was not failing and the intention was in fact to drop data from the continuous aggregates.

Reply:
We actually can't check `cascade_to_materializations` at this point as we have to do this update after we complete the rc1 update.  So that parameter will no longer exist by the time we do this update.

Reply:
But I guess it must be possible to check this during the update process at some point. I guess it is fine to not do this and just disable all policies, even those that weren't failing and "did the right thing". But we must decide that this is the semantics we'd like to have and then describe the behavior in the migration doc so that users can be advised to take appropriate action.

Reply:
I think the best approach at this point is to raise this notice and change all tables that have this conflict.  It will be annoying for any users who are intentionally doing this, but I can't actually think of a valid use case where a user should intentionally be doing this (it seems just dropping the aggregated chunk would be the solution they'd want to be using).